tests/const_/mod.rs

```diff
@@ -107,6 +108,39 @@ fn const_nothing() {
     assert_eq!(actual.out, "nothing");
 }
 
+#[rstest]
+#[case(&["const x = not false", "$x"], "true")]
+#[case(&["const x = false", "const y = not $x", "$y"], "true")]
+#[case(&["const x = not false", "const y = not $x", "$y"], "false")]
+fn const_unary_operator(#[case] inp: &[&str], #[case] expect: &str) {
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, expect);
+}
+
+#[rstest]
+#[case(&["const x = 1 + 2", "$x"], "3")]
+#[case(&["const x = 1 * 2", "$x"], "2")]
+#[case(&["const x = 4 / 2", "$x"], "2")]
+#[case(&["const x = 4 mod 3", "$x"], "1")]
+#[case(&["const x = 5.0 / 2.0", "$x"], "2.5")]
+#[case(&[r#"const x = "a" + "b" "#, "$x"], "ab")]
+#[case(&[r#"const x = "a" ++ "b" "#, "$x"], "ab")]
+#[case(&[r#"const x = [1,2] ++ [3]"#, "$x | describe"], "list<int>")]
+#[case(&[r#"const x = 0x[1,2] ++ 0x[3]"#, "$x | describe"], "binary")]
+#[case(&[r#"const x = 0x[1,2] ++ [3]"#, "$x | describe"], "list<any>")]
+fn const_binary_operator(#[case] inp: &[&str], #[case] expect: &str) {
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, expect);
+}
```

Comment:
Can you throw in a few error cases also?

I was wondering if division by zero becomes a parser error since the const eval is happening at parse time.

Reply:
Yes it's a parse-time error and shown like this:

<img width="505" alt="divzeroerror" src="https://github.com/nushell/nushell/assets/1991933/e0f3adad-0817-4278-8025-d62eb4f0451e">

tests for error cases is added.